source/parser.hera

```diff
@@ -621,13 +621,13 @@ AssignmentExpression
   # NOTE: Try to match a single line assignment expression before matching newline then assignment expression
   SingleLineAssignmentExpression
   # TODO: Ideally this wouldn't be needed.
-  __ AssignmentExpressionTail
+  ( EOS _? ) AssignmentExpressionTail
   # NonPipelineAssignmentExpression
 
 NonPipelineAssignmentExpression
   # NOTE: Try to match a single line assignment expression before matching newline then assignment expression
   NonPipelineSingleLineAssignmentExpression
-  __ NonPipelineAssignmentExpressionTail
+  ( EOS _? ) NonPipelineAssignmentExpressionTail
```

Comment:
I wonder if `NotDedented` would make sense here? I think maybe not, the idea is to break indentation here. But thought I'd mention the possibility.

Overall, looks like a good approach!